ThirdParty/ecec/src/params.c

```diff
@@ -22,38 +22,47 @@
 #define ECE_HEADER_STATE_END_VALUE 9
 #define ECE_HEADER_STATE_INVALID_HEADER 10
 
+// Extracts an unsigned 32-bit integer in network byte order.
+static inline uint32_t
+ece_read_uint32_be(const uint8_t* bytes) {
+  uint32_t value = bytes[3];
+  value |= (uint32_t) bytes[2] << 8;
+  value |= (uint32_t) bytes[1] << 16;
+  value |= (uint32_t) bytes[0] << 24;
+  return value;
+}
+
 // A linked list that holds name-value pairs for a parameter in a header
 // value. For example, if the parameter is `a=b; c=d; e=f`, the parser will
 // allocate three `ece_header_pairs_t` structures, one for each ;-delimited
 // pair. "=" separates the name and value.
 typedef struct ece_header_pairs_s {
+  struct ece_header_pairs_s* next;
   // The name and value are pointers into the backing header value; the parser
   // doesn't allocate new strings. Freeing the backing string will invalidate
   // all `name` and `value` references. Also, because these are not true C
   // strings, it's important to use them with functions that take a length, like
   // `strncmp`. Functions that assume a NUL-terminated string will read until
   // the end of the backing string.
   const char* name;
-  size_t nameLen;
   const char* value;
+  size_t nameLen;
   size_t valueLen;
-  struct ece_header_pairs_s* next;
 } ece_header_pairs_t;
 
 // Initializes a name-value pair node at the head of the pair list. `head` may
 // be `NULL`.
 static ece_header_pairs_t*
 ece_header_pairs_alloc(ece_header_pairs_t* head) {
-  ece_header_pairs_t* pairs =
-    (ece_header_pairs_t*) malloc(sizeof(ece_header_pairs_t));
+  ece_header_pairs_t* pairs = malloc(sizeof(ece_header_pairs_t));
```

Comment:
I like to include the cast to make things more explicit. But thinking about it again I don't think it's actually necessary.